src/databricks/labs/ucx/source_code/graph.py

```diff
@@ -318,8 +317,8 @@ def __repr__(self):
         return f"Dependency<{self.path}>"
 
     @property
-    def lineage_str(self):
-        return self._lineage_str
+    def lineage(self) -> Lineage:
```

Comment:
```suggestion
    def lineage(self) -> list[LineageAtom]:
```
a lot of the problems could be overcome by changing the return type of the output

Reply:
indeed, done